File: lib/models/AudioTrack.dart
class AudioTrack {
  final String artist;
  final String title;
  final String thumbnail;
  final int length;
  final int position;
  final int queuePosition;
  final String videoID;
  final bool isStream;

  AudioTrack({
    required this.artist,
    required this.title,
    required this.thumbnail,
    required this.length,
    required this.position,
    required this.queuePosition,
    required this.videoID,
    required this.isStream,
  });

  String get durationFormatted {
    final minutes = (length / 60).floor();
    final seconds = length % 60;
    return '$minutes:${seconds.toString().padLeft(2, '0')}';
  }
}

File: android/app/src/main/java/com/zenyfh/zenmusic/AudioTrack.java
package com.zenyfh.zenmusic;

import org.jetbrains.annotations.NotNull;

import java.util.HashMap;
import java.util.Map;

public class AudioTrack {
    private final String artist;
    private final String title;
    private final String thumbnail;
    private final int length;   // seconds
    private final String streamUrl;
    private final int position;       // seconds
    private final boolean isStream;
    private final Map<String, Object> mapObject;
    private final int queuePosition;

    public AudioTrack(String artist, String title, String thumbnail, int length, int position, int queuePosition, String streamUrl, boolean isStream) {
        this.artist = artist;
        this.title = title;
        this.thumbnail = thumbnail;
        this.length = length;
        this.position = position;
        this.queuePosition = queuePosition;
        this.streamUrl = streamUrl;
        this.isStream = isStream;
        this.mapObject = createMapObject();
    }

    private Map<String, Object> createMapObject() {
        Map<String, Object> mapObject = new HashMap<>();
        mapObject.put("artist", artist);
        mapObject.put("title", title);
        mapObject.put("thumbnail", thumbnail);
        mapObject.put("length", length);
        mapObject.put("position", position);
        mapObject.put("queuePosition", queuePosition);
        mapObject.put("streamUrl", streamUrl);
        mapObject.put("isStream", isStream);
        return mapObject;
    }

    public Map<String, Object> getMapObject() {
        return mapObject;
    }

    @NotNull
    @Override
    public String toString() {
        return String.format("AudioTrack(%s - %s)", artist, title);
    }
}
